### ISSUE TITLE
how to create the dataset ？

### ISSUE THREAD

**Issue by Johnson-yue** [yongqyu/TCN#1]
Hi，I am interested in your work。 I want to run your code ， but I failed . because of missing make datasets code!!!
I have three question about you dataset format:

1. get_loader function:
    In your **main** function, any data_loader used [get_loader()](https://github.com/yongqyu/TCN/blob/5c537cc692c4889aff5e64410fd48242f324f99b/data_loader.py#L104) , It get return data from [EncImageFolder()](https://github.com/yongqyu/TCN/blob/5c537cc692c4889aff5e64410fd48242f324f99b/data_loader.py#L114) and what is the meaning of return datas?
 such as : 
         if I have three images it including  typeface_1_char_1, typeface_1_char_2, typeface_2_char_1, and I want to generate **typeface_2_char_2.  Could you tell me the three images which is _src_img_  and which is _char_img_ ,and _style_img_ ？？

2. Do you train full network with only one picture??  because  you train code use [ImageFolder()](https://github.com/yongqyu/TCN/blob/5c537cc692c4889aff5e64410fd48242f324f99b/data_loader.py#L112) function

Finally,  Would you release your code of making dataset?

**Comment by yongqyu** [comment]
Hi, Johnson yue.
This is Yonggyu Park.

First of all thank you for your interest.
And please understand that datasets can not be published due to typeface
issue issues.

Below is an answer to your question.
1. src_img is an anchor image. In your example, it is typeface_1_char_1.
And style_img is an image with different style with anchor. That is,
typeface_2_char_1. For the same reason, char_img is typeface_1_char_2.

And TCN does not generate typeface_2_char_2 through typeface_1_char_1, but
generates typeface_1_char_2.
Also note that EncImageFolder() is used only for Encoder pre-training, and
ImageFolder() is used for main training.

2. The ImageFolder() function returns two images(input and target).
Therefore, it is correct to use only one image in train and validation.

If you have any questions, please feel free to contact me.
Thank you.

2018년 11월 12일 (월) 오후 7:06, Johnson yue <notifications@github.com>님이 작성:

**Comment by Johnson-yue** [comment]
I think there are some __Ambiguity__ in my example. In my example, the hypothesis is I have typeface_1_char_N (where N is 0-27,000 maybe)  and I only have typeface_2_char_1, so I want generate other typeface_2_char_M( where M is 1-27,000 maybe).

If follow this hypothesis, TCN do not should generate typeface_1_char_2 ,is right?

**Comment by yongqyu** [comment]
You are right.

And I omit one, file names are like 11_42.png. 11 is the typefaece index,
and 42 is the character index. It would be better to refer to it.

Thank you.

2018년 11월 13일 (화) 오전 11:10, Johnson yue <notifications@github.com>님이 작성:

**Comment by Johnson-yue** [comment]
In your EncImageFolder(),  the **char_tg_idx** and **sytle_tg_idx**, it is not used in pretrain ! Yes?

Btw, you use triplet loss when accuracy of each encoder higher than 80% (but paper is 90%, it does not matter), I want to know how much improve for performance with triplet loss?? Do you check it?

**Comment by yongqyu** [comment]
As you say, char_tg_idx and style_tg_idx are not used. I have added theses,
just in case.

No specific comparative experiments were performed in the second question.
However, if triplet loss is not applied, there is a problem that each
encoder extracts only typeface or character information. So I added this.

Thanks.
2018년 11월 13일 (화) 오후 12:03, Johnson yue <notifications@github.com>님이 작성:

**Comment by Johnson-yue** [comment]
ok, thank you , I will using keras-tf reimplement it ,and my image is 256*256, did you use 256 image size for testing or training?

**Comment by yongqyu** [comment]
We have experimented with various image sizes. It was difficult to extract
the style information throw DeepCNN at size smaller than a certain size.
256 * 256 could not be implemented due to hardware limitations.

2018년 11월 13일 (화) 오후 12:36, Johnson yue <notifications@github.com>님이 작성:

**Comment by Johnson-yue** [comment]
Hi，It is me, again. Did you train with charset is larger ? such as 6763, so your generator input channel should be 2x512+2x6763. It is so big for update , and as your paper the [mix layer](https://github.com/yongqyu/TCN/blob/69fefbd231f4b330c03d99a3a522704f3e7e40be/model.py#L208) weights shape should be (2x512+2x6763, 2x512+2x6763, 1, 1), It is so large parameter. 

And I think your Network have a lot of redundant parameters ,so if you want to increase input image size， you should modify your network.

Did you update your network structure ?

**Comment by yongqyu** [comment]
We use 1,000 character classes and agree that the mix layer has a large
number of parameters.
We are currently working on a revision model, and if you describe
the duplicate parameters, we reflect them..
Thank you for your attention.

2018년 11월 14일 (수) 오후 7:04, Johnson yue <notifications@github.com>님이 작성:

**Comment by Johnson-yue** [comment]
Hi, I am back ,again~.  I look your [train code](https://github.com/yongqyu/TCN/blob/69fefbd231f4b330c03d99a3a522704f3e7e40be/solver.py#L293) and [data_load for train](https://github.com/yongqyu/TCN/blob/69fefbd231f4b330c03d99a3a522704f3e7e40be/data_loader.py#L54). The following is my understanding：

1.  your data_load return _Training data_  and the format is (src_img, trg_img, style_idx, char_idx, char_trg_idx, style_trg_idx)   For convenience, let me mark the return data again.
                     * Mark  **src_img** as S1C1 it means  Style_1_char_1.
                     * Mark **trg_img** as S1C2 it means  Style_1_char_2.
                     * Mark **style_idx** as S1  it means  Style_1_index.
                     * Mark **char_idx** as C1  it means  char_1_index.
                     * Mark **char_trg_idx** as C2  it means  char_2_index.
                     * Mark **style_trg_idx** as S1  it means  Style_1_index.    I think you code function is choice the same style and the different char .

2. my question is Are you not use Style_2_char_N ???? So your train data is _subset of characters_ (as your paper Abstract) , is right?  Do not need other style data ??

**Comment by Johnson-yue** [comment]
Other Question , If I have a subset of characters (such as  500 chars), and I want to 6763 chars。
As your pipeline, I can use any one char of the subset **Generate** other  (6763 - 500=6263) chars.
Finally, I have 500 * 6263 chars for choice , it is not a good idea

**Comment by yongqyu** [comment]
Hi Johnson yue.

First of all, I did not understand correctly about the recent question.
If you have 500 characters out of 6,763 character set that correspond to a
single typeface. In this situation, is it the question of 500 * 6,263
choice to make the remaining 6,263 characters? That's right.

TCN is a model designed to complete typeface. However, this problem can be
caused by creating a single character at a time. We are still in the
process of being modified. The 'subset of characters' in the paper is now
just 'one character'. The model using the subset will proceed to future
work. I will update my paper and github soon.

Thank you.

2018년 11월 15일 (목) 오후 4:09, Johnson yue <notifications@github.com>님이 작성:

**Comment by Johnson-yue** [comment]
As my opinion， the prefect model is  input full subset of character （not only one character）， and output is remaining character。

But you model is input only one character and output is remaining character。So I have to choice which output data is the best, but it is not bug. This is just my opinion。

And I looking forward to your update！！！

**Comment by yongqyu** [comment]
I agree with your opinion.
I will try to update it as soon as possible.
Thank you very much.

2018년 11월 15일 (목) 오후 4:48, Johnson yue <notifications@github.com>님이 작성:
